metronome/checkpointing/service/src/io/iohk/metronome/checkpointing/service/CheckpointingService.scala

```diff
@@ -280,7 +303,7 @@ object CheckpointingService {
       ledgerStorage: LedgerStorage[N],
       blockStorage: BlockStorage[N, CheckpointingAgreement],
       viewStateStorage: ViewStateStorage[N, CheckpointingAgreement],
-      pushCheckpointFn: CheckpointCertificate => F[Unit]
+      interpreterClient: InterpreterRPC[F]
```

Comment:
I know this is a draft, but as I [commented](https://github.com/input-output-hk/metronome/pull/53#discussion_r666075541) before I thought `apply` would get the local connection manager and create the client directly, so that it can pass the `ServiceRPC` instance that has access to things like the `mempool` created further down.